releasetasks/test/firefox/__init__.py

```diff
@@ -45,6 +45,23 @@ def do_common_assertions(graph):
             assert "routes" in task
             rel_routes = [r.startswith("index.releases.") for r in task["routes"]]
             assert len(rel_routes) >= 2, "At least 2 release index routes required"
+
+            if task["provisionerId"] == "buildbot-bridge":
+                assert "treeherder" not in task["extra"], \
+                    "remove treeherder from {}: {}".format(task_name, task)
+                assert "treeherderEnv" not in task["extra"], \
+                    "remove treeherderEnv from {}: {}".format(task_name, task)
+                assert not any([r.startswith("tc-treeherder") for r in task["routes"]]), \
+                    "remove treeherder routes from {}: {}".format(task_name, task)
+
+            if task["provisionerId"] == "aws-provisioner-v1":
+                assert "treeherder" in task["extra"], \
+                    "add treeherder to {}: {}".format(task_name, task)
+                assert "treeherderEnv" in task["extra"], \
+                    "add treeherderEnv to {}: {}".format(task_name, task)
+                assert len([r for r in task["routes"] if r.startswith("tc-treeherder")]) == 2, \
+                    "{} has to have 2 treeherder routes: {}".format(task_name, task)
+
```

Comment:
Yeah... The current approach is against the whole idea of unittests to hit all asserts.